File: leetcode/js/8.Atoi.js
class ATOI{
   static atoi(s){
      let st = s.trim()
      if(st == null || st.length ==0){
        return 0
      }
      let res=0
      let isNegative=false
      let startIndex=0
      
      if(st[0]=='+' || st[0]=='-'){
         startIndex+=1
      }
 
      if(st[0]=='-'){
         isNegative=true
      } 
      
      for(let i=startIndex;i<st.length;i++){
        if(st[i]<'0'||st[i]>'9'){
          break;
        }
        let digit = parseInt(st[i])
        res = res*10+digit
      }
 
      if(isNegative){
        res= -1 * res
      }
 
      if(res > Math.pow(2,31) - 1)
         return Math.pow(2,31)-1
 
      if(res < Math.pow(-2 ^ 31) + 1)
         return Math.pow(-2,31)+1
 
      return res
   }
 }
 
 console.log(ATOI.atoi(" 42"))

File: leetcode/kotlin/8.Atoi.kt
class ATOI{
    companion object{
        fun atoi(s:String):Int{
            var res:Double=0.0
            var str = s.trim()
            var isNegative:Boolean = false
            var startIndex:Int=0
            if(str[0]=='+'||str[0]=='-'){
                startIndex++
            }
            if(str[0]=='-'){
                isNegative = true
            }
            for(i: Int in startIndex until str.length){
                if(str[i]<'0'||str[i]>'9'){
                    break;
                }
                var digit:Int = (str[i] - '0').toInt()
                res = res *10 + digit
            }
            if(isNegative){
                res = -res
            }
            if(res > Int.MAX_VALUE){
                return Int.MAX_VALUE
            }else if(res<Int.MIN_VALUE){
                return Int.MIN_VALUE
            }
            return res.toInt()
        }
    }
}

fun main(){
  print(ATOI.atoi(" 42"))
}